Amiberry 5.6.6

rahim opened the pull request:
This bumps Amiberry from 5.6.4 -> 5.6.6.

Release notes:
https://github.com/BlitterStudio/amiberry/releases/tag/v5.6.5
https://github.com/BlitterStudio/amiberry/releases/tag/v5.6.6

Notably this changes the default vsync behaviour, which should fix https://github.com/BlitterStudio/amiberry/issues/1189  – render corruption when launching from EmulationStation on Pi5. (That was my own motivation, but I'm yet to confirm it has the desired outcome).

libportmidi-dev [has been added as a build dependency](https://github.com/BlitterStudio/amiberry/commit/e52bd6f44d776affc7017192b7ef691c496a3420).

cmitu commented:
Thanks for the PR. Which systems has this been tested on ?

You also want to include `cmake` as depenency and squash the commits into a single one. Is there a way to disable MUNT (MT32 emulation) for weaker systems (Pi0/1/2/3) ?

rahim commented:
Hi @cmitu 

I should have warned I'm not familiar with either project, I've been enjoying tinkering to try and get things behaving for my setup and largely cargo culted this upgrade PR from earlier work. I thought it might save someone a few minutes of wrangling to generate the new patchfile given the upstream Makefile had been updated.

> Which systems has this been tested on?

The only thing I've tested is that compiling amiberry v5.6.6 succeeds on Pi5 (8GB) running on 64 bit Pi OS, 8KB page size, RetroPie installed manually following the guide.

I don't have earlier generation Pis readily available to test with.

> You also want to include cmake as depenency and squash the commits into a single one.

👍 That's easily done, I did wonder about cmake, but presumed it must have been required earlier, so perhaps covered elsewhere.

> Is there a way to disable MUNT (MT32 emulation) for weaker systems (Pi0/1/2/3) ?

My lack of knowledge is exposed by this – that was carried verbatim from the changes in amiberry, I'm not sure of the background to that choice.

rahim commented:
> Notably this changes the default vsync behaviour, which should fix https://github.com/BlitterStudio/amiberry/issues/1189 – render corruption when launching from EmulationStation on Pi5. (That was my own motivation, but I'm yet to confirm it has the desired outcome).

The changes in 5.6.6 didn't resolve my issue here either as it happens, I need to do some more digging in to that.

rahim commented:
I've squashed and added `cmake`.

Right now I've not rebuilt with that, but I can't see that creating a problem.

As best I can tell, there's currently not support for compiling without MUNT. Beyond compilation cost, it's not clear to me it would incur runtime cost unless users choose to lean on that feature?

Given the upgrade doesn't resolve the problem I was hoping it would, and given the limited testing picture, merging perhaps represents more trouble than it's worth right now.

cmitu commented:
> As best I can tell, there's currently not support for compiling without MUNT. Beyond compilation cost, it's not clear to me it would incur runtime cost unless users choose to lean on that feature?

 Yes, it doesn't seem so. As for the cost - it's just removing extra compilation steps and possible support questions from users which don't have powerful enough devices to run it.

> Given the upgrade doesn't resolve the problem I was hoping it would, and given the limited testing picture, merging perhaps represents more trouble than it's worth right now.

The upgrade cannot solve the issue you had, since it's not an Amiberry bug - it's a Linux kernel bug which will be solved in raspberrypi/linux. What's fixed in Amiberry is the _Vsync_ not being read and applied from the configuration file at start-up (option which works around the kernel bug). 

Tested on Pi3 Buster/Pi4 Bookworm and it builds and runs ok. I think we can merge this.

rahim commented:
> The upgrade cannot solve the issue you had, since it's not an Amiberry bug - it's a Linux kernel bug which will be solved in raspberrypi/linux. What's fixed in Amiberry is the Vsync not being read and applied from the configuration file at start-up (option which works around the kernel bug).

Yes, I eventually got to that conclusion. Great to have a workaround and that the real fix is on the way.

> Tested on Pi3 Buster/Pi4 Bookworm and it builds and runs ok. I think we can merge this.

Excellent, thanks for the steer!

joolswills commented:
Thank you.

rahim commented:
Thank you, great project!
